include/swift/Frontend/ParseableInterfaceModuleLoader.h

```diff
@@ -134,10 +134,11 @@ class ParseableInterfaceModuleLoader : public SerializedModuleLoaderBase {
   std::string PrebuiltCacheDir;
 
   std::error_code findModuleFilesInDirectory(
-    AccessPathElem ModuleID, StringRef DirPath, StringRef ModuleFilename,
-    StringRef ModuleDocFilename,
-    std::unique_ptr<llvm::MemoryBuffer> *ModuleBuffer,
-    std::unique_ptr<llvm::MemoryBuffer> *ModuleDocBuffer) override;
+      AccessPathElem ModuleID, StringRef DirPath, StringRef ModuleFilename,
+      StringRef ModuleDocFilename,
+      std::unique_ptr<llvm::MemoryBuffer> *ModuleBuffer,
+      std::unique_ptr<llvm::MemoryBuffer> *ModuleDocBuffer,
+      std::string *ParseableInterface) override;
```

Comment:
This isn't really what I meant; I figured the PIML should be able to wait until the module has been loaded and then add this itself. Since that doesn't really work in the current architecture I still think we have a problem.

@harlanhaskins, thoughts?

Reply:
Hmm, where would the PIML add this? Also, it might be simpler to just add a field in the serialized module itself containing this path, and have `ParseableInterfaceBuilder` pass it down through `SerializationOptions`, no?

Reply:
@jrose-apple Yeah, as far as I understand the PIML isn't involved in loading the serialized AST, it triggers the build of the .swiftmodule and returns a MemoryBuffer that we then pass to loadAST().

@harlanhaskins We can already read the path out of the input control block; my original patch did that. A separate field would be redundant, but arguably a little cleaner.

Reply:
That's how it's factored now, but…

Okay, if you both think a field is the way to go, then let's do that. But let's make it a proper field—the base name usually *won't* match up because it'll be a target triple rather than the module name.

Reply:
Done.